File: 0047-permutations-ii/0047-permutations-ii.cpp
class Solution {
public:
    void solve(vector<int>& nums,unordered_map<int,int>&mp, vector<int>& temp,vector<vector<int>>& results) {

        if(temp.size()==nums.size()){ results.push_back(temp); return; }
        for(auto& pair:mp) {
            if(pair.second==0) continue;
            pair.second-=1;
            temp.push_back(pair.first);
            solve(nums,mp,temp,results);
            temp.pop_back();
            pair.second+=1;
        }
    }
    vector<vector<int>> permuteUnique(vector<int>& nums) {
        unordered_map<int,int> mp;
        for(auto num:nums) mp[num]++;
        vector<int> temp;
        vector<vector<int>> results;
        solve(nums,mp,temp,results);
        return results;
        
    }
};

File: 0047-permutations-ii/0047-permutations-ii.py
class Solution:
    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        finalAns = []
        nums.sort()
        def solve(index,temp):
            if index==len(nums): 
                finalAns.append(list(temp))
                return
            for i in range(len(nums)):
                if i>0 and nums[i]==nums[i-1]: continue
                if nums[i]!=11:
                    tempVar = nums[i]
                    temp.append(nums[i])
                    nums[i]=11
                    solve(index+1,temp)
                    temp.pop()
                    nums[i] = tempVar
        solve(0,[])
        return finalAns
        